json_diff_test.go

```diff
@@ -427,3 +427,235 @@ func TestShouldReturnChangesWhenAObjectIsMissing(t *testing.T) {
 		t.Error("Test failed. Expected", expected, "but returned", actual)
 	}
 }
+
```

Comment:
You could write a test to check if the input values were not changed by the `DiffWithValues` method, ensuring the copy is being made.

Reply:
done!